recommender-system/README.md

```diff
@@ -1 +1,135 @@
-# Recommender system
\ No newline at end of file
+# Recommender system
+
+## AWS Setup:
+We want to use Python and the NumPy module in an AWS Lambda function. In this objective, the only solution found was to create a dockerized environnement.
+
+### Container setup with Docker for Python
+1. On your local machine, create a project directory for your new function.
+2. In your project directory, add a file named app.py containing your function code, then add your function handler code to the app.py function. Example 
+```
+import numpy
+
+def handler(event, context):
+    return np.zeros((10,10))
+    
+```
+
+3. In your project directory, add a file named requirements.txt. List each required library as a separate line in this file.
+4. Create a Dockerfile like so:  
+Install any dependencies under the ${LAMBDA_TASK_ROOT} directory alongside the function handler to ensure that the Lambda runtime can locate them when the function is invoked.
+```
+FROM public.ecr.aws/lambda/python:3.9
+
+# Copy function code
+COPY app.py ${LAMBDA_TASK_ROOT}
+
+# Install the function's dependencies using file requirements.txt
+# from your project folder.
+
+COPY requirements.txt  .
+RUN  pip3 install -r requirements.txt --target "${LAMBDA_TASK_ROOT}"
+
+# Set the CMD to your handler (could also be done as a parameter override outside of the Dockerfile)
+CMD [ "app.handler" ]
+```
+
+5. Create an Amazon ECR
+
+6. Build your Docker image with `docker build -t hello-world .`
+7. Authenticate the Docker CLI to your Amazon ECR registry.  
+`aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin 123456789012.dkr.ecr.us-east-1.amazonaws.com`
+
+8. Create a repository in Amazon ECR directly with the ECR interface (easier).
+
+9. Tag your image to match your repository name, and deploy the image to Amazon ECR using the docker push command.  
+`docker tag  hello-world:latest 123456789012.dkr.ecr.us-east-1.amazonaws.com/hello-world:latest`  
+`docker push 123456789012.dkr.ecr.us-east-1.amazonaws.com/hello-world:latest`
+
+10. To update the version :  
+`docker build -t 123456789012.dkr.ecr.us-east-1.amazonaws.com/hello-world:latest`  
+`docker push 123456789012.dkr.ecr.us-east-1.amazonaws.com/hello-world:latest`
+
+### Create a Database with DynamoDB
+
+1. Create a table:
+    - Key partitioning: Chain
+2. Possibly create elements within the table if required
+3. Get the ARN (required to connect to the Lambda function)
```

Comment:
The Lambda function > Which one? Pas de **the** pour éviter l'ambiguïté, sinon pointe vers la fonction en question